src/test/controller/BaseController.ts

```diff
@@ -94,6 +98,19 @@ export class BaseController {
     return new AWSMock.S3();
   };
 
+  /**
+   * Create a mock authenticated user
+   * */
+  async createAuthenticatedUser(): Promise<void> {
+    // create user
+    const user = await (this.mockFirebaseAdmin as any).createUser({
+      email: 'test@gmail.com',
+      password: 'testpassword',
```

Comment:
I'm just wondering the part `(this.mockFirebaseAdmin as any)` . What does this do?

Reply:
Ohya good question, before we had it like [this](https://github.com/ubclaunchpad/PromoPal-backend/blame/bc94c9f1e17189b367b201bf82ef1fb056c2f95f/src/test/controller/UserController.test.ts#L48-L51), and `mockFirebaseAdmin` was already type `any`. I've just casted it back to `any` because our mocking library seems to complain if it isn't `any`.  I think this is because the mock-firebase-admin doesnt have supported typescript, but let me double check if I can fix this